test/testplot.jl

```diff
@@ -0,0 +1,102 @@
+using SparseArrays
+using Random
+
+const ALL_PLOT_PRIMITIVES = [
+    Scatter, Lines, LineSegments, Makie.Text, Mesh, MeshScatter, Image, Heatmap,
+    Surface, Volume, Voxels
+]
+const ALL_PLOT_TYPES = vcat(ALL_PLOT_PRIMITIVES, [
+    # skipped Annotations, Axis3D
+    ABLines, Arc, Arrows, Band, Makie.BarPlot, Bracket, Contourf, Contour, Contour3d,
+    DataShader, Errorbars, Rangebars, HLines, VLines, HSpan, VSpan, Pie, Poly,
+    RainClouds, ScatterLines, Series, Spy, Stairs, Stem, StreamPlot, TimeSeries,
+    Tooltip, Tricontourf, Triplot, VolumeSlices, Voronoiplot, Waterfall, Wireframe,
+    BoxPlot, CrossBar, Density, QQPlot, QQNorm, ECDFPlot, Hexbin, Hist, Violin
+])
+
+"""
+    testplot!(scenelike, PlotType[; kwargs...])
+
+Creates a sample plot of type `PlotType` with random data in the given `scenelike`.
+Any keyword arguments are passed to the plot function.
+
+This is meant to simplify creating throw-away plots for testing attributes.
+`ALL_PLOT_TYPES` may also be useful.
+"""
+function testplot!(scene, ::Type{PlotType}, kwargs...) where {PlotType <: Plot}
+    CT = Makie.conversion_trait(PlotType)
+    f = Makie.MakieCore.plotfunc!(PlotType)
+
+    if CT === PointBased() || PlotType <: Union{Makie.Text, ABLines, BarPlot,
```

Comment:
Can we make this multiple dispatch based, maybe so it returns aplotspec that can be decomposed to args and kwargs and then plotted?

That way it's easy to extend as we get more recipes, or for external recipes we may want to test.

Reply:
Yea I guess multiple dispatch would be better if another library wants to reuse and extend the Code for testing. If it's just used here I don't think multiple dispatch matters much.

I'm not sure if PlotSpec is a good idea. It's calling `to_value` on kwargs/attributes and I'd assume it circumvents parts of the normal plot pipeline. It should probably be tested separately

Reply:
Makes sense, maybe just returning a tuple `(; args, kwargs)` might work?